File: server/src/main/java/com/ecommerce/ecommercerestapi/Entity/Comment.java
package com.ecommerce.ecommercerestapi.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "user_comment")
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @ManyToOne(cascade = { CascadeType.PERSIST, CascadeType.MERGE })
    private User user;

    @ManyToOne(cascade = { CascadeType.PERSIST, CascadeType.MERGE })
    private Product product;

    @Column(name = "quantity_star")
    private Integer quantityStar;

    private String message;
}

File: client/src/interfaces/Comment.ts
import { Product } from "./Product";
import { User } from "./User";

export interface CommentReq {
  user: {
    id: number;
  };
  product: {
    id: number;
  };
  message: string;
  quantityStar: number;
}

export interface CommentRes {
  id?: number;
  user: User;
  product: Product;
  message: string;
  quantityStar: number;
}
